server/protocols/jmap/src/main/java/org/apache/james/jmap/DownloadServlet.java

```diff
@@ -49,12 +50,50 @@
 
     private static final String ROOT_URL = "/";
     private static final Logger LOGGER = LoggerFactory.getLogger(DownloadServlet.class);
+    private static final String TEXT_PLAIN_CONTENT_TYPE = "text/plain";
 
     private final MailboxSessionMapperFactory mailboxSessionMapperFactory;
+    private final SimpleTokenManager simpleTokenManager;
 
     @Inject
-    @VisibleForTesting DownloadServlet(MailboxSessionMapperFactory mailboxSessionMapperFactory) {
+    @VisibleForTesting DownloadServlet(MailboxSessionMapperFactory mailboxSessionMapperFactory, SimpleTokenManager simpleTokenManager) {
         this.mailboxSessionMapperFactory = mailboxSessionMapperFactory;
+        this.simpleTokenManager = simpleTokenManager;
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
+        String pathInfo = req.getPathInfo();
+        if (Strings.isNullOrEmpty(pathInfo) || pathInfo.equals(ROOT_URL)) {
+            resp.setStatus(SC_BAD_REQUEST);
+        } else {
+            respondAttachmentAccessToken(getMailboxSession(req), blobIdFrom(pathInfo), resp);
+        }
+    }
+
+    private void respondAttachmentAccessToken(MailboxSession mailboxSession, String blobId, HttpServletResponse resp) {
+        try {
+            if (! attachmentExists(mailboxSession, blobId)) {
```

Comment:
I am not sure coding convention allows a space between ! and attachementExists.
WDYT ?